lib/yaml_normalizer.rb

```diff
@@ -23,12 +24,7 @@ def dump(file, data)
   end
 
   def handle_hash(hash)
-    copy = hash.each_value { |value| handle_value(value) }.dup
-    hash.clear
-    copy.keys.sort.each do |key|
```

Comment:
Does this no longer sort by key? Can we bring that back to be more deterministic?

Reply:
> Does this no longer sort by key? Can we bring that back to be more deterministic?

Ah, I didn't realize that was being enforced by this code. I'll bring it back.

Reply:
> > Does this no longer sort by key? Can we bring that back to be more deterministic?
> 
> Ah, I didn't realize that was being enforced by this code. I'll bring it back.

Restored in the rebased 4f07697.

And some related benchmarking, for posterity: https://gist.github.com/aduth/cede3176fc6194959a3be22b7d49ccc1

Reply:
thanks for adding the sorting tests too!

Reply:
> thanks for adding the sorting tests too!

Except, now that I look at them again, they're wrong 😄 

This ain't sorted 😬  :

https://github.com/18F/identity-idp/blob/4f07697b77dba4143b34f84d8e8d5353fea759d1/spec/lib/yaml_normalizer_spec.rb#L50-L59

Though I'd guess it's more to do with the fact that `{ a: 1, b: 2 } == { b: 2, a: 1 }`, so assertion needs to be changed to account for keys being sorted.

Reply:
Better: 4441891

Reply:
🤦  maybe do a `.to_a` on the hashes, that will linearize them

Reply:
`to_json` seemed to retain the order as well, though admittedly I'm unsure if this is guaranteed. A bit more direct and the failure message reads decently-well.